### PR TITLE
Replace shuttle with http-async

### PR DESCRIPTION
**Pull request by anuragsoni**
I'm switching the shuttle implementation to use `http_async` which was published to opam recently. `http_async` uses shuttle internally.

image

The image is from a benchmark run under docker on my machine with `RUN_DURATION=20`

I don't think I can catch up to `eio` when it uses io-uring in polling mode, but I'm glad to see that the refactoring I did in shuttle (I switched to a higher level simpler API), hasn't resulted in my http server falling too far behind others :-)

### CODE REVIEW

File: run_benchmarks.sh [hunk 1]
```diff
@@ -3,16 +3,16 @@ set -xe
...
 run_duration="${RUN_DURATION:-60}"
 
-export GOMAXPROCS=24
-export COHTTP_DOMAINS=24
-export HTTPAF_EIO_DOMAINS=24
-export RUST_CORES=24
+export GOMAXPROCS=1
+export COHTTP_DOMAINS=1
+export HTTPAF_EIO_DOMAINS=1
+export RUST_CORES=1
```
Comment (anuragsoni) [inline run_benchmarks.sh:9]: https://github.com/ocaml-multicore/retro-httpaf-bench/blob/6d7aaeac4a7d00be05b5f81d5ceead14de4facac/run_benchmarks_mult.sh exists for multicore benchmarks, so I switched the number of domains back to 1 in this script.

**Comment by talex5** [comment]
CI is failing with:
```
# File "cohttp-async/src/input_channel.mli", line 13, characters 16-34:
# 13 | val to_reader : Core_kernel.Info.t -> t -> Reader.t Deferred.t
#                      ^^^^^^^^^^^^^^^^^^
# Error: Unbound module Core_kernel
```

**Comment by anuragsoni** [comment]
@talex5 the Cohttp errors might be related to the 0.15 series of Janestreet libraries. Core_kernel was renamed to Core. I’m a little surprised to see cohttp_async in the errors as from what I can tell there are no cohttp_async examples in this repository.

**Comment by anuragsoni** [comment]
Cohttp's current trunk works well with 0.15 series of janestreet libraries (https://github.com/mirage/ocaml-cohttp/pull/867). I'm guessing the eio version of cohttp that's included in this repository as a submodule will need to be updated to sync with the new cohttp commits.

**Comment by talex5** [comment]
I made a PR (#26) to fix the `cohttp-lwt-unix` build to avoid that. It looks like `httpaf-shuttle-async` has a similar problem, but maybe rebasing this PR on it will fix both.

**Comment by anuragsoni** [comment]
Rebasing this PR on top of #26 should indeed help. The error in #26 is related to a breaking change in `shuttle`. Another option that should work is to pin shuttle to an older release as part of #26 `shuttle.0.3.1` should work.

**Comment by anuragsoni** [comment]
@talex5 I have rebased this on top of #26 , and the build succeeds now.

**Comment by talex5** [comment]
Thanks!

**Pull request merged**